R/teal_data-get_code.R

```diff
@@ -0,0 +1,50 @@
+#' Get Code from `teal_data`
+#'
+#' Get code from [`teal_data`]. The extraction of the code limited to specific objects can be performed with the usage
+#' of `names` parameter. If input `teal_data@verified` is `FALSE` the returned code is pre-pended with a warning
+#' stating that the object was not verified.
+#'
+#' @param x [`teal_data`] object.
+#' @param names (`character`) vector of object names to return the code for.
+#' @param deparse (`logical`) whether return deparsed form of a call.
+#' @return (`character`) code of import and preparation of data for `teal` application.
+#' @examples
+#'
+#' tdata1 <- teal_data()
+#' tdata1 <- within(tdata1, {
+#'   a <- 1
+#'   b <- a^5
+#'   c <- list(x = 2)
+#' })
+#' get_code(tdata1)
+#' datanames(tdata1) <- c("a", "b", "c")
+#' get_code(tdata1, "a")
+#' get_code(tdata1, "b")
+#'
+#' tdata2 <- teal_data(x1 = iris, code = "x1 = iris")
+#' get_code(tdata2)
+#' get_code(verify(tdata2))
+#' @aliases get_code,teal_data-method
+#' @export
+setGeneric("get_code", function(x, ...) standardGeneric("get_code"))
```

Comment:
Please use generic from `teal.code`. Overloading generic causes problems with S4 methods table

Reply:
Example what can happen:

<img width="375" alt="Skärmavbild 2023-11-22 kl  09 03 57" src="https://github.com/insightsengineering/teal.data/assets/6959016/34228a69-db33-4820-89cf-d2600ff8bd15">

Reply:
Yeah, I had an issue with that yesterday and didnt understand the cause. Thanks for the hint

Reply:
I had a warning yesterday that get_code generic is missing `...` so I can not add parameters to get_code other than those existing in generic (`object` and `deparse`). I added `...` to the generic, but it looks like I should add it to `teal.code` and not `teal.data`. Added an issue and PR for that in `teal.code` https://github.com/insightsengineering/teal.code/pull/169